File: packages/ui-compose/ui/src/main/kotlin/com/kinetixui/ui/Skeleton.kt
package com.kinetixui.ui

import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.graphicsLayer
import androidx.compose.ui.res.dimensionResource

/**
 * KinetixSkeleton — mirrors `packages/ui/src/components/skeleton.tsx`
 * (`animate-pulse rounded-md bg-muted`). Has no intrinsic size, same as the
 * React version, which is sized entirely by the caller's `className`; here
 * that's the caller's `modifier` (e.g. `Modifier.fillMaxWidth().height(20.dp)`).
 */
@Composable
fun KinetixSkeleton(modifier: Modifier = Modifier) {
    val colors = KinetixColorScheme.current
    val transition = rememberInfiniteTransition(label = "KinetixSkeleton")
    val alpha by transition.animateFloat(
        initialValue = 1f,
        targetValue = 0.5f,
        animationSpec = infiniteRepeatable(
            animation = tween(durationMillis = 1000),
            repeatMode = RepeatMode.Reverse,
        ),
        label = "KinetixSkeletonAlpha",
    )
    Box(
        modifier = modifier
            .graphicsLayer { this.alpha = alpha }
            .clip(RoundedCornerShape(dimensionResource(R.dimen.radius_md)))
            .background(colors.muted),
    )
}

File: packages/ui-swiftui/Sources/KinetixUI/Skeleton.swift
//
// Skeleton.swift — KinetixSkeleton.
//
// Mirrors packages/ui/src/components/skeleton.tsx (`animate-pulse
// rounded-md bg-muted`). No intrinsic size — the caller sizes it with a
// `.frame(...)`, the same as the React version being sized by `className`.
//

import SwiftUI

public struct KinetixSkeleton: View {
    @Environment(\.kinetixColors) private var colors
    @State private var pulsed = false

    public init() {}

    public var body: some View {
        RoundedRectangle(cornerRadius: 8, style: .continuous) // radius/md
            .fill(colors.muted)
            .opacity(pulsed ? 0.5 : 1)
            .animation(
                .easeInOut(duration: 1).repeatForever(autoreverses: true),
                value: pulsed
            )
            .onAppear { pulsed = true }
    }
}
